app/scripts/modules/core/src/application/nav/NavRoute.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { UIRouterContextComponent } from '@uirouter/react-hybrid';
+import { UISref } from '@uirouter/react';
+
+import { NavCategory } from './NavCategory';
+import { ApplicationDataSource } from '../service/applicationDataSource';
+import { Application } from '../../application';
+
+export interface INavRouteProps {
+  category: ApplicationDataSource;
+  isActive: boolean;
+  app: Application;
+}
+
+export const NavRoute = ({ app, category, isActive }: INavRouteProps) => (
+  <UIRouterContextComponent key={category.key}>
```

Comment:
The `<UIRouterContextComponent/>` is only necessary if the thing rendering this component is AngularJS (typically because the React component has been wrapped using `react2angular`).  If the parent components are all React and rendered inside a React `UIView`, then this kludge isn't necessary.  I'm not sure where this is going to be used, but I wanted to point that out.

Reply:
Ok I will remove